File: php/sumoftwo.php
<?php
// https://www.algoexpert.io/questions/Two%20Number%20Sum

/* 
First version
A loop within a loop, not ideal
*/
function sumOfTwo($array, $target){
    $len = count($array);
    $result = [];
    for($i = 0; $i < $len; $i++){
        for($j = $i+1; $j < $len; $j++){
            if($array[$i] + $array[$j] == $target){
                $result = [$array[$i], $array[$j]];
                var_dump($result);
                return $result;
            }
        }
    }
    return $result;
}
$array1 = [3, 5, -4, 8, 11, 1, -1, 6];
$target1 = 10;
sumOfTwo($array1, $target1);
$array2 = [4, 6, 1, -3];
$target2 = 3;
sumOfTwo($array2, $target2);
?>

File: python/sumoftwo.py
# https://www.algoexpert.io/questions/Two%20Number%20Sum

# first version
# not using python ability to search efficiently here
# resulting in loops iterating through the whole array
# added a quick end if a pair is found
def twoNumberSum(array, targetSum):
	output = []
	array_copy = array.copy()
	for element in array:
		array_copy.remove(element)
		for element2 in array_copy:
			#print(f"{element} - {element2}")
			if element + element2 == targetSum:
				output.append(element)
				output.append(element2)
			if len(output) > 0: # end the function early if pair is found
				break
		if len(output) > 0: #end the function early if a pair is found
				break
	return output